lib/typed.js

```diff
@@ -147,6 +147,7 @@ exports.symbol = function(string) {
   if (string !== null) {
     assert.string(string, "string");
   }
+  string = string.replace(/\u0000/g, "");
```

Comment:
could you move this one line up into the `if (string !== null) { }` to fix the broken test?